### PR TITLE
Undo TRT-1378: Revert for "OCPCLOUD-2278: Add kube-rbac-proxy container & ensure metrics are only available via HTTPS"

### PR DESCRIPTION
**Pull request by racheljpg**
This is a PR to add back the reverted changes and fix the issues that were the cause of the reversion. My first commit adds the changes back, the second changes the placeholder image. Thanks!

### CODE REVIEW

File: manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml [hunk 1]
```diff
@@ -139,3 +166,10 @@ spec:
         hostPath:
           path: /etc/kubernetes
           type: Directory
+      - configMap:
+          name: kube-rbac-proxy
+        name: auth-proxy-config
+      - name: cloud-controller-manager-operator-tls
+        secret:
+          secretName: cloud-controller-manager-operator-tls
+          optional: true
```
Comment (JoelSpeed) [inline manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml:175]: Nit, should have new line character at the end

Comment (racheljpg) [reply manifests/0000_26_cloud-controller-manager-operator_11_deployment.yaml]: Done!

**Comment by JoelSpeed** [comment]
/payload-job periodic-ci-openshift-hypershift-release-4.15-periodics-e2e-aws-ovn-conformance

**Pull request merged**